### DISCUSSION TITLE
how to rollback scheduler.schedule(instance)

### DISCUSSION THREAD

**Discussion by GithubRyze** [kagkarlsson/db-scheduler#536] [Q&A]
My business have two step like belows:
1.  insert  record to TABLE A 
2.  if step 1 success then create a scheduler
so i must use Transactional commit. but i used spring @Transactional and transactionTemplate are invalide.   step  1 failed but  step 2 always created a scheduler.

RecordRepository extends JpaRepository,  
```
@Transactional(transactionManager = "transactionManager",rollbackFor = Exception.class)
  public String createTask(record Record) {
   recordRepository.save(record )
   scheduler.schedule(instance);
}

```

```
scheduler = Scheduler
        .create(ds, httpTaskScheduler)
        .pollingInterval(Duration.ofSeconds(1L))
        .registerShutdownHook()
        .statsRegistry(new ScheduledLogStatsPlainRegistry(meterRegistry, scheduledLogDomainService))
        .heartbeatInterval(Duration.ofSeconds(30L))
        .shutdownMaxWait(Duration.ofSeconds(30L))
        .build();
    scheduler.start();
```

```
@Bean
  public PlatformTransactionManager transactionManager(EntityManagerFactory entityManagerFactory) {
    JpaTransactionManager transactionManager = new JpaTransactionManager();
    transactionManager.setEntityManagerFactory(entityManagerFactory);
    transactionManager.setDataSource(dataSource);
    return transactionManager;
  }
```

**Comment by kagkarlsson** [comment]
It should participate in transactions if you are using the boot-autostarter. Are you? (see [code](https://github.com/kagkarlsson/db-scheduler/blob/master/db-scheduler-boot-starter/src/main/java/com/github/kagkarlsson/scheduler/boot/autoconfigure/DbSchedulerAutoConfiguration.java#L215))

If you are instantiating the Scheduler or the SchedulerClient yourself, you probably need to wrap it in a `TransactionAwareDataSourceProxy`.
